src/main/kotlin/org/jetbrains/research/ml/codetracker/tracking/TaskFileHandler.kt

```diff
@@ -82,11 +76,9 @@ object TaskFileHandler {
             logger.info("Project $project is already added or set to be added")
             return
         }
-        if (PluginServer.serverConnectionResult == ServerConnectionResult.SUCCESS) {
+        SurveyUiData.programmingLanguage.currentValue?.let {
```

Comment:
This check is not enough for creating task files, because if the user is on the SurveyPane they can change the language again. I think we should call initProject if the programming language is set and the user started task solving.